crypto/armor.go

```diff
@@ -193,16 +212,33 @@ func UnarmorDecryptPrivKey(armorStr string, passphrase string) (privKey cryptoty
 	return privKey, header[headerType], err
 }
 
-func decryptPrivKey(saltBytes []byte, encBytes []byte, passphrase string) (privKey cryptotypes.PrivKey, err error) {
-	key, err := bcrypt.GenerateFromPassword(saltBytes, []byte(passphrase), BcryptSecurityParameter)
-	if err != nil {
-		return privKey, errorsmod.Wrap(err, "error generating bcrypt key from passphrase")
+func decryptPrivKey(saltBytes []byte, encBytes []byte, passphrase string, kdf string) (privKey cryptotypes.PrivKey, err error) {
+	// Key derivation
+	var key []byte
+	switch kdf {
+	case KDFArgon2:
+		key = argon2.IDKey([]byte(passphrase), saltBytes, argon2Time, argon2Memory, argon2Threads, argon2KeyLen)
+	default:
+		key, err = bcrypt.GenerateFromPassword(saltBytes, []byte(passphrase), BcryptSecurityParameter)
+		if err != nil {
+			return privKey, errorsmod.Wrap(err, "error generating bcrypt key from passphrase")
+		}
+		key = crypto.Sha256(key) // Get 32 bytes
 	}
 
-	key = crypto.Sha256(key) // Get 32 bytes
+	// Symetrhic decryption
+	aead, err := chacha20poly1305.New(key)
+	if len(encBytes) < aead.NonceSize() {
+		return privKey, errorsmod.Wrap(err, "error generating aead cypher from key")
+	}
+	nonce, privKeyBytesEncrypted := encBytes[:aead.NonceSize()], encBytes[aead.NonceSize():] // Split nonce and ciphertext.
+	privKeyBytes, err := aead.Open(nil, nonce, privKeyBytesEncrypted, nil)                   // Decrypt the message and check it wasn't tampered with.
```

Comment:
For ciphertexts that were encrypted before this change (that used bcrypt to derive the key, and xsalsa20symmetric to encrypt), you need to use xsalsa20symmetric to decrypt.
Maybe the symmetric decryption could also be in the switch case? (and it would probably be good to add a comment explaining that both the symmetric encryption algorithm and the KDF were updated at the same time, and that's why the decryption algorithm depends on the switch case which is over the KDF field)